src/pytest_mypy_testing/plugin.py

```diff
@@ -55,13 +54,15 @@ def __init__(
         self.mypy_item = mypy_item
         for mark in self.mypy_item.marks:
             self.add_marker(mark)
+        if have_xdist:
+            self.add_marker(pytest.mark.xdist_group("mypy"))
```

Comment:
This could be done separately, but targets the same overall goal of running mypy once. Without this, `pytest-xdist` could potentially build a mypy cache for every worker. This puts all mypy jobs on the same worker (provided the user passes `--dist loadgroup` to pytest), so only one cache is built.